src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -86,15 +90,22 @@ void generate_chunk_pruning_statistics(const std::shared_ptr<Chunk>& chunk) {
 
 void generate_chunk_pruning_statistics(const std::shared_ptr<Table>& table) {
   const auto chunk_count = table->chunk_count();
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(chunk_count);
+
   for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
-    const auto chunk = table->get_chunk(chunk_id);
+    const auto generate_statistics = [&, chunk_id]() {
+      const auto chunk = table->get_chunk(ChunkID{chunk_id});
 
-    if (!chunk || chunk->is_mutable()) {
-      continue;
-    }
+      if (!chunk || chunk->is_mutable()) {
+        return;
+      }
 
-    generate_chunk_pruning_statistics(chunk);
+      generate_chunk_pruning_statistics(chunk);
+    };
+    jobs.emplace_back(std::make_shared<JobTask>(generate_statistics));
```

Comment:
Will not be measurable but we could put that all into a single line. Make shared on existing data adds another allocation (data block and control block not adjacent) and then we need to copy with emplace instead of creating in place